services/pub/pub-points.service.js

```diff
@@ -0,0 +1,58 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+
+const documentation = `<p>A measure of quality. This includes several dimensions of quality such as code style, platform support, and maintainability.</p>`
+
+const keywords = ['dart', 'flutter']
+
+const schema = Joi.object({
+  grantedPoints: Joi.number().min(0).max(130).required(),
+  maxPoints: Joi.number().min(130).max(130).required(),
+}).required()
+
+const title = 'Pub Points'
+
+export default class PubPoints extends BaseJsonService {
+  static category = 'rating'
+
+  static route = { base: 'pub/points', pattern: ':packageName' }
+
+  static examples = [
+    {
+      title,
+      keywords,
+      documentation,
+      namedParams: { packageName: 'analysis_options' },
+      staticPreview: {
+        label: 'points',
+        message: '130/130',
+        color: 'brightgreen',
+      },
+    },
+  ]
+
+  static defaultBadgeData = { label: 'points' }
+
+  static render({ grantedPoints, maxPoints, packageName }) {
+    return {
+      label: 'points',
+      message: `${grantedPoints}/${maxPoints}`,
+      color: 'brightgreen',
```

Comment:
Given there is a ceiling on this one, rather than hard-coding the color we can assign a color based on some kind of scale using `floorCount()` https://github.com/badges/shields/blob/9186f0d2620f09adfd94f555d3174eb4102d4a64/services/color-formatters.js#L35-L47 like we do with test coverage, for example: https://github.com/badges/shields/blob/9186f0d2620f09adfd94f555d3174eb4102d4a64/services/color-formatters.js#L53-L55
Is there any kind of community norm around what is considered a "good" score we could lean on here? We shouldn't make them all `brightgreen` even if the score is like 4/130 or whatever

Reply:
For now, I have added the blue color for pub points as there are no such norms for points.

Reply:
With this one it seems like the scale is more esoteric. Clearly bigger numbers are better, but is 20 "Documentation" points and 0 "Platform Support" points exactly the same as 10 "Documentation" points and 10 "Platform Support" points. Maybe lets leave this one as blue/informational.

Reply:
I have made some changes and assigned colors based on the percentage of their granted points and max points. i.e.
```js
floorCount((grantedPoints / maxPoints) * 100, 40, 60, 80)
```